mycore-iview2/src/main/java/org/mycore/iview2/iiif/MCRIVIEWIIIFImageImpl.java

```diff
@@ -289,5 +305,36 @@ private MCRTiledPictureProps getTiledPictureProps(Path tiledFile) throws MCRIIIF
         }
         return tiledPictureProps;
     }
+    
+    private MCRTileInfo createTileInfo(String identifier) throws MCRIIIFImageNotFoundException {
+        MCRTileInfo tileInfo = null;
+        String[] splittedIdentifier = identifier.split("/", 2);
+        switch (splittedIdentifier.length) {
+        case 1:
+            tileInfo = new MCRTileInfo(null, identifier, null);
+            break;
+        case 2:
+            tileInfo = new MCRTileInfo(splittedIdentifier[0], splittedIdentifier[1], null);
+            break;
+        default:
+            throw new MCRIIIFImageNotFoundException(identifier);
+        }
+        return tileInfo;
+    }
+
+    private void checkTileFile(String identifier, MCRTileInfo tileInfo, Path tileFilePath)
+        throws MCRAccessException, MCRIIIFImageNotFoundException {
+        if (!Files.exists(tileFilePath)) {
+            throw new MCRIIIFImageNotFoundException(identifier);
+        }
+        if (tileInfo.getDerivate() != null) {
+            if (!MCRAccessManager.checkPermission(tileInfo.getDerivate(), MCRAccessManager.PERMISSION_READ)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=4800529)